tests/Functional/Schema/SQLServerSchemaManagerTest.php

```diff
@@ -404,4 +404,31 @@ public function testPkOrdering(): void
         self::assertEquals('colB', $columns[0]);
         self::assertEquals('colA', $columns[1]);
     }
+
+    public function testListSchemaNames(): void
+    {
+        //Schemas db_* with schema_id>=16384 are backward compatible schemas:
+        //https://learn.microsoft.com/en-us/sql/relational-databases/security/authentication-access/ownership-and-user-schema-separation?view=sql-server-ver16#built-in-schemas-for-backward-compatibility
+        //and should't be listed because creating DDL object is not possible on such schemas
+        //Te same page contains information about schemas: dbo, guest, sys, INFORMATION_SCHEMA
+        //they are system fixed and can't be dropped or re-created so listing them for SchemaDiff doesn't make sense
+        $schemas = $this->schemaManager->listSchemaNames();
+
+        self::assertThat($schemas, self::logicalAnd(
+            self::assertNotContains('dbo',$schemas),                //schema_id=1
+            self::assertNotContains('guest',$schemas),              //schema_id=2
+            self::assertNotContains('INFORMATION_SCHEMA',$schemas), //schema_id=3
+            self::assertNotContains('sys',$schemas),                //schema_id=4
+
+            self::assertNotContains('db_owner',$schemas),           //schema_id=16384
```

Comment:
I believe just checking one of the names is sufficient. If we don't use the magic constants, we can rely on the SQL Server to maintain the metadata of all schemas properly.